vdom/src/com/vdom/core/Game.java

```diff
@@ -2398,7 +2395,7 @@ protected void initPlayerCards() {
         tradeRouteValue = 0;
         if (isCardInGame(Cards.tradeRoute)) {
             for (AbstractCardPile pile : piles.values()) {
-                if ((pile.card().is(Type.Victory)) && !pile.card().is(Type.Knight) && !pile.isBlackMarket()) {
+                if ((pile.topCard().is(Type.Victory)) && !pile.topCard().is(Type.Knight) && !pile.isBlackMarket()) {
```

Comment:
Condition this on the type of the placeholder card and whether the pile is a supply pile instead of this logic (since the placeholder/randomizer determines the piles' type).